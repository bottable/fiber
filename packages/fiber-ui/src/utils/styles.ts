import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
