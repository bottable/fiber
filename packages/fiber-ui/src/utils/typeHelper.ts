export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
