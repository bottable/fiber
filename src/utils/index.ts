export * from './typeHelper'
export * from './mergeProps'
