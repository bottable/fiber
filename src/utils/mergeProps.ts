import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
