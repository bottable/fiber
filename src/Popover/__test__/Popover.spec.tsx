import { Popover } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Popover', () => {
  it('renders correctly', () => {
    const popover = render(<Popover />)

    expect(popover).toMatchSnapshot()
  })
})
