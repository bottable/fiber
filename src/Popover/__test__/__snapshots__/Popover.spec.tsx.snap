// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover renders correctly 1`] = `
<div
  className="sc-fzpans jDFSfP"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <button
    className="sc-fzqAbL sc-fzoYHE eHCbTQ"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span>
      Hover me
    </span>
  </button>
  <div
    className="sc-fzplWN eLdScE"
  >
    <div
      className="sc-AxhUy bgNvMX"
      size="sm"
    >
      <div
        className="sc-AxmLO gzOyEY"
        size="sm"
      />
    </div>
  </div>
  <div
    className="sc-fznyAO hXbUPA"
  />
</div>
`;
